Library/Homebrew/download_strategy.rb

```diff
@@ -303,7 +303,11 @@ def ext
     # We can't use basename_without_params, because given a URL like
     #   https://example.com/download.php?file=foo-1.0.tar.gz
     # the extension we want is ".tar.gz", not ".php".
-    Pathname.new(@url).extname[/[^?]+/]
+    url_pathname = Pathname.new(@url)
+    until ext = url_pathname.extname[/[^?]+/]
```

Comment:
This should probably handle the case where the extname never matches and/or `dirname` fails.

Reply:
@kabel Ping on this

Reply:
I addressed this in https://github.com/Homebrew/brew/pull/3657/commits/0254605cd52b1bb92ffe33114c671cc111c5a687 by checking the return value of dirname.

Reply:
@kabel That's not sufficient. For example, if `url_pathname` ends up as `/` then this will be an infinite loop. I think it's worth changing the approach so this loop can never be infinite.

Reply:
I did test it out through irb. I could add the root `/` as another escape condition, but these are going to be URLs that we're dealing with. So the top-level domain would (in most cases where it going that far) be returned as the extension. If the URL didn't have a domain (`file:///`), dirname would still default to `.`. I guess it can't hurt to be even more robust.

Reply:
@kabel I'm sorry if I'm not being explicit enough but I'm just not happy with a change that adds an infinite loop if the return values are ever something you do not expect. Instead of making this loop indefinitely until there's an escape condition instead consider splitting on `/` and iterating over that or another approach that makes this a bounded loop.